SC_redesign/Crop_and_Soil/soil/carbon_cycling/carbon_cycle.py

```diff
@@ -0,0 +1,347 @@
+                                                                                  soil_mass)
+            soil_passive_carbon_fraction = self._determine_soil_passive_carbon_fraction(layer.passive_carbon_amount,
+                                                                                        soil_mass)
+            layer.soil_overall_carbon_fraction = self._determine_soil_overall_carbon_fraction(
+                soil_active_carbon_fraction,
+                soil_slow_carbon_fraction,
+                soil_passive_carbon_fraction)
+            layer.total_soil_carbon_amount = self._determine_total_soil_carbon_amount(layer.active_carbon_amount,
+                                                                                      layer.slow_carbon_amount,
+                                                                                      layer.passive_carbon_amount)
+            total_plant_carbon_CO2_loss = self._determine_total_plant_carbon_CO2_loss(
+                layer.plant_metabolic_active_carbon_loss,
+                layer.plant_structural_active_carbon_loss,
+                layer.plant_structural_slow_carbon_loss
+            )
+            total_soil_carbon_CO2_loss = self._determine_total_soil_carbon_CO2_loss(
+                layer.soil_metabolic_active_carbon_loss,
+                layer.soil_structural_active_carbon_loss,
+                layer.soil_structural_slow_carbon_loss
+            )
+            layer.total_decomposition_carbon_CO2_lost = self._determine_total_decomposition_carbon_CO2_lost(
+                layer.active_carbon_to_slow_loss,
+                layer.slow_carbon_co2_lost_amount,
+                layer.passive_carbon_co2_lost_amount
+            )
+            layer.total_carbon_CO2_lost = self._determine_total_carbon_CO2_lost(
+                total_plant_carbon_CO2_loss,
+                total_soil_carbon_CO2_loss,
+                layer.total_decomposition_carbon_CO2_lost)
+
+    @staticmethod
+    def _determine_soil_volume(layer_thickness: float, field_size: float) -> float:
+        """This method calculates the soil volume
+
+        Parameters
+        ----------
+        layer_thickness: float
+            thickness of soil layer (mm)
+        field_size: float
+            Size of the field (ha)
+
+        Returns
+        -------
+        float
+            soil volume (cubic meters)
+
+        References
+        -------
+        pseudoode_soil S.6.D.1
+        """
+        return (layer_thickness * field_size * HECTARES_TO_SQUARE_MILLIMETERS) * CUBIC_MILLIMETERS_TO_CUBIC_METERS
+
+    @staticmethod
+    def _determine_soil_mass(bulk_density: float, soil_volume: float) -> None:
+        """This method calculates the mass of soil
+
+        Parameters
+        ----------
+        bulk_density: float
+            bulk density of the soil layer (Mg per cubic meter)
+        soil_volume: float
+            soil volume (cubic meters)
```

Comment:
Spell out the units here to avoid confusion (double-check that this is correct).

```suggestion
        bulk_density: float
            bulk density of the soil layer (Megagrams per cubic meter)
        soil_volume: float
            soil volume (cubic meters)
```